### ISSUE TITLE
WebPlatform Web Components F2F 2018

### ISSUE THREAD

**Issue by TakayoshiKochi** [WICG/webcomponents#713]
At TPAC, we agreed to have another F2F around February 2018, at somewhere not in the US, so everyone can join. Let's plan the next F2F! (last update Feb. 20, 2017)

Date: Mar. 5-6 (Mon-Tue), 2018 10am - 6pm
Venue: Google Tokyo Office (Roppongi, Tokyo, Japan)
For detailed logistics see [w3c's meetings page](https://github.com/w3c/WebPlatformWG/blob/gh-pages/meetings/18-03-Web-components.md).

- Agenda (1.5hour x 4 slots / day)
  - Day 1
    - Agenda building + unconference planning (30min) / Go through existing issues (1)
    - Go through existing issues (2)
    - [Template instantiation](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md)
    - Unconference 1
  - Day 2
    - [HTML modules](#645)
    - [Scoped Custom Elements registries](https://github.com/w3c/webcomponents/issues/716) / [Declarative Shadow DOM](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Declarative-Shadow-DOM.md)
    - Unconference 2
    - Unconference 3

**Comment by annevk** [comment]
Some thoughts:

* Multiple days seems good.
* Let's decide well in advance which days to focus on existing issues versus new functionality. That makes it easier to plan for folks with less time/budget.
* Let's make sure to extend invitations to React, Ember, Vue.js, ? well in advance as well. Representation from Angular has historically not been a problem, but to succeed we'll need more folks to weigh in.
* If we manage to get the various frameworks to attend I'd like us to have some unconference sessions for problems with the web platform in general, not restricted to web components.

**Comment by annevk** [comment]
Since email addresses are hard to come by: @yyx990803 @wycats @tomdale @sophiebits @gaearon @sebmarkbage would you all (or someone you know who can represent the JavaScript framework you work on) be interested in meeting up in Toronto next year February primarily to discuss extensions to the DOM and HTML in browser implementations ("Web Components"), but not necessarily limited to that.

Our rough plan is to have a day to go through existing issues, and one or more days to discuss new ideas, such as [HTML modules](https://github.com/w3c/webcomponents/issues/645) and [HTML Template Instantiation](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md), and learn from you all what the pain points are with the platform.

**Comment by TakayoshiKochi** [comment]
@justinfagnani @azakus @kevinpschaaf from the Polymer team (they were at TPAC) may also be interested in this.

**Comment by yyx990803** [comment]
For the record, I am particularly interested in HTML modules and HTML Template Instantiation and would be willing to attend the days that are focusing on these.

**Comment by tomalec** [comment]
Could we bring up customized built-ins/extending native elements/custom elements with different parsing contexts to the agenda? 

I would say, >430 comments, hundreds of thumbs-ups, many participants in the single issue and even more if we count related issues, proves the need for it.

**Comment by annevk** [comment]
Pointer? Is there a concrete proposal with some implementer support?

**Comment by domenic** [comment]
My understanding is that customized built-ins, as specced, have 3/4 implementer support. So they may not be a fruitful topic for discussion; instead web developers can bring up their concerns with their Apple representatives directly at another venue (e.g. in said centithread).

**Comment by tomdale** [comment]
I would definitely be interested in attending. My email address, if it makes things easier: tom@tomdale.net.

**Comment by TakayoshiKochi** [comment]
Updated the agenda in OP.
Tentatively set the schedule for 3 days - which means it starts on Monday, Tuesday, or Wednesday
(to Wednesday, Thursday, or Friday, respectively).
Also I'd like to ping @travisleithead and @rniwa if anyone from MS and Apple can join.

**Comment by rniwa** [comment]
Do we really need three days? I don't think I can quite afford to be there for three days.

**Comment by hober** [comment]
Two days sounds good. I think 2/19–20 work for me.

**Comment by JanMiksovsky** [comment]
2/19/2018 is President's Day, a federal holiday in the U.S. Many people have that day off, and some people stretch the holiday weekend by a day or so.

As it turns out, in the corner of the country where I live (Seattle), many schools have that entire week off as a mid-winter break. My children happen to attend two such schools, so while I'd _really_ like to make it to this meeting, I couldn't attend a meeting held the week of 2/19.

Obviously any given time will be inconvenient for some people, so the group should decide what works well for a quorum of people. But if this could be deferred to the following week or early March, that'd help people taking time off around that national holiday.

**Comment by annevk** [comment]
@rniwa @hober my idea was that we'd have two structured days focused around existing issues and new proposals, and one day that's a bit more exploratory. If we stick to that then I think those that want to take less time can figure out for themselves which days they're interested in. And hopefully nobody falls in the category of wanting to skip new proposals...

FWIW, I can do a week later (26 - 28, 27 - 1, or 28 - 2) in theory.

**Comment by rniwa** [comment]
If it's on the week of Feb 26th, then I can't do Monday, Tuesday, or Wednesday.

**Comment by annevk** [comment]
Here's a doodle: https://doodle.com/poll/nry9ut6n3g7x2cge. I suggest we pick based on the result of that doodle at the end of next week (so please please fill it in when you read this message) so everyone has plenty of time to make arrangements.

**Comment by TakayoshiKochi** [comment]
Thanks @annevk for setting up the doodle!
I'm okay for any of these 3 weeks, but considering many are from the US and the holiday, the week of Feb. 19 was not a good option, and considering @rniwa is a must for the meeting, Mar. 1-2 or
the week of Mar.5 would currently looks the stronger candidate.

If we were to squeeze the schedule into 2 days, maybe we can prepare for going through the existing bugs beforehand, and shorten what need to be done in F2F, then we can still have time for discussing new stuff?  E.g. On each day if we have 4 slots (1.5-2hrs?), how about doing like this?

1st day
- contentious bugs resolution
- template instantiation
- template processing
- buffer / unconf 1

2nd day
- HTML modules
- HTML modules (cont.)
- unconf 2
- buffer / remaining stuff

We need to have pre-meeting prep for identifying really contentious parts, and bring concrete proposals or use cases beforehand, but it would make the meeting more productive, I believe.

**Comment by rniwa** [comment]
Yeah, something like for two days sounds good. I also like the first week of March.

**Comment by domenic** [comment]
I've filled out the Doodle and have a conflicting event on the 1st and 2nd. First week of March is sounding more promising.

I would prefer to avoid meetings on Mondays or Fridays since they would then require traveling on a weekend.

**Comment by rniwa** [comment]
I'd prefer not having this meeting in the middle of a week since I have way too many other commitments, and flying during a week day would mean that I'd have to work over weekends to compensate for it.

**Comment by TakayoshiKochi** [comment]
Thanks for putting your dates on [doodle](https://doodle.com/poll/nry9ut6n3g7x2cge).
As @annevk already said, let's close this at the end of this week (Sat. Dec. 16) to decide the date.
Note that you don't have to put your name/date if you are flexible, putting your name is not mandatory for joining the meeting.

So far people are good for 2 days conf, but I see some schedule conflict.
I'd try best to accommodate all preferences, so please speak up!

**Comment by justinfagnani** [comment]
I think first week of March works for me too, I'm checking.

I'd like to throw scoped custom element registrations on the agenda, if possible: https://github.com/w3c/webcomponents/issues/716

@TakayoshiKochi is there a difference between "template instantiation" and "template processing"? By "template processing" do you mean just focusing on the built-in template processor?

**Comment by TakayoshiKochi** [comment]
> is there a difference between "template instantiation" and "template processing"? By "template processing" do you mean just focusing on the built-in template processor?

No, I just split the HTML Template Instantiation into 2 parts somewhat arbitrarily, guessing that it would need time for 2 slots.

**Comment by travisleithead** [comment]
> Also I'd like to ping @travisleithead and @rniwa if anyone from MS and Apple can join.

We'll try to have someone from Microsoft be there regardless of the dates--either myself or (hopefully) one or two of my colleagues; haven't figured that out just yet...

**Comment by robwormald** [comment]
I’ll attend from the Angular team, and am flexible on dates.

**Comment by developit** [comment]
I'll be happy to attend for Preact!

**Comment by rniwa** [comment]
By the way, while Toronto is a lovely city there might be a snow storm & long delays for flights in February & March so I would throw in Tokyo & London as possible alternatives (assuming Google / Mozilla's respective offices can accommodate us) although that might make traveling harder for some U.S.-based individuals.

**Comment by sophiebits** [comment]
Closing the loop from the React side: I don't think we'll make it to this unless there's something specific you'd like input on – we're not actively working on anything in this space. Thanks for thinking of us!

**Comment by TakayoshiKochi** [comment]
I have never been to Toronto so I have no idea how terrible to be there at the season... but I believe Ryosuke for his experience.  @annevk @smaug---- do you have any idea where it would be in Europe?
If it will be in Tokyo, I'm happy to work on securing meeting rooms at Google Tokyo.
Let's decide on location asap for everyone to schedule.

**Comment by smaug----** [comment]
Mozilla's office in London may not have good meeting space. The office in Berlin would be better.

**Comment by prushforth** [comment]
> Could we bring up customized built-ins/extending native elements/custom elements with different parsing contexts to the agenda?

I second this.  I would like to add that there is a general need for the ability to create 'speculative polyfills' for unknown elements, which autonomous elements don't support.  Example &lt;foo is="new-foo"&gt;

>I would say, >430 comments, hundreds of thumbs-ups, many participants in the single issue and even more if we count related issues, proves the need for it.

+1

> My understanding is that customized built-ins, as specced, have 3/4 implementer support. So they may not be a fruitful topic for discussion; instead web developers can bring up their concerns with their Apple representatives directly at another venue (e.g. in said centithread).

Is Chrome waiting on Safari to implement this?  If so they will be waiting a long time, I think.  Better to follow Mozilla's example and implement based on about as good a consensus as you are likely to get.  
Because the webcomponents speculative polyfill for built-ins are blocked on Chrome, apparently, so Web developers can't even vote with their feet.

At the very least, I think there's something to discuss.  My .02 cents.

**Comment by domenic** [comment]
> Is Chrome waiting on Safari to implement this?

No.

**Comment by adamdbradley** [comment]
I would also be interested in attending for https://stenciljs.com/ and https://ionicframework.com/

**Comment by caridy** [comment]
@TakayoshiKochi I spoke to @domenic earlier today, we could probably host it in Paris since we will not be hosting any tc39 meeting in 2018 (and we still have the budget). We have a very nice facility there in Paris and the weather should not be a problem I believe. Just let me know if that helps and we can coordinate the details. /cc @diervo

**Comment by annevk** [comment]
I think Tokyo might be slightly better for some from Mozilla (quite a few of us are in Taipei and plan to attend), though from that perspective Paris is also better than Toronto.

**Comment by TakayoshiKochi** [comment]
Considering the doodle votes, week of Feb. 19 and Feb. 26 has significant collisions - so the week of Mar.5 (sorry @tomdale).  And to avoid moving on the weekend, let's go for Mar. 6-7 (Tue-Wed).
Also, for location, to avoid severe weather / traffic trouble, let's settle on Tokyo.

Date: Mar. 6-7 (Tue-Wed)
Location: Google Tokyo office (Roppongi, Tokyo, Japan)

Unless we hear strong oppositions for date and location in this week, let's finalize it.

**Comment by TakayoshiKochi** [comment]
@caridy Thanks for the offer!  we may still have to choose plan B, but the current plan A is Tokyo.

**Comment by rniwa** [comment]
Ugh... can we do Mon-Tue instead? Tue-Wed schedule would likely make it impossible for me to attend in person if it were to happen in Tokyo. I can't be out of office for three straight days as I've previously stated.

**Comment by annevk** [comment]
Asking people to travel on the weekend is also rather bad though.

**Comment by rniwa** [comment]
Well, if we can't do Mon-Tue or Thu-Fri, then I probably need to call in instead of being there in person. I'm sorry but I have managerial obligations from mid-February to early April. I can only attend this meeting in person if it consumes at most two business days during that period. Tue-Wed would have been okay if it were held in Toronto because then I can fly there on Mon evening and fly back on Wed evening but if it's in Tokyo, there isn't really a way.

**Comment by TakayoshiKochi** [comment]
@rniwa I see, but you proposed Tokyo or London... which were both impossible?
Also, even if Mon-Tue in Tokyo, it means you have to fly back on Tue early night (and get back in SFO on Tue), which means you can't attend full day 2?  And even worse, Thu-Fri, you have to leave SFO in Tue, arrive Tokyo in Wed. night - then you will use 4 straight week days?

**Comment by rniwa** [comment]
I've stated this earlier: https://github.com/w3c/webcomponents/issues/713#issuecomment-351233942

The only reason I've suggested London/Tokyo was that I was concerned about the storm in Toronto but with the assumption that the meeting would take place on Mon/Tue or Thu/Fri (Thu/Fri would be bad in Tokyo too because I can't leave on Wednesday in PST and arrive in time for Thursday morning in JST). If we need to have it on Tue/Wed or Wed/Thi, then Toronto is probably the only option we have.

**Comment by rniwa** [comment]
Mon-Tue works because I can leave Tokyo on Wed evening and arrive in the morning of Wed in California. If we're doing Thu-Fri, I would have had to leave Wed mid day and arrive there Thursday morning

**Comment by TakayoshiKochi** [comment]
Yeah, I saw the comment but I thought you would be fine for either Tokyo or London from the other comment.

Okay, then 2 options:
- Toronto, Tue-Wed
- Tokyo Mon-Tue

If in Toronto, for people who are prepared for bad weather can arrive on Sunday for one-day buffer, and for busy people can also have tight flight schedule?

I guess both options have pros/cons for everyone, so I'd like to wait for reactions.
Please give your votes on the following 2 comments with your reaction.

**Comment by TakayoshiKochi** [comment]
Toronto, Tue-Wed (Mar. 6-7)

**Comment by TakayoshiKochi** [comment]
Tokyo, Mon-Tue (Mar. 5-6)

**Comment by TakayoshiKochi** [comment]
I'll close the vote tomorrow night (~24hrs away) and make a decision. 
It looks like @rniwa has preference to Toronto (against Tokyo), but assuming he's okay if it's Mon-Tue (although I sympathize the tough schedule) and if the reactions indicate more people comfortable with Tokyo, we'd go with Tokyo.

**Comment by smaug----** [comment]
Tokyo on Mon-Tue is fine to me.

But I did tentatively reserve the large meeting space at Mozilla's Toronto office, Tue-Wed (Mar. 6-7)

**Comment by caridy** [comment]
Paris, Tue-Wed (Mar. 6-7) - We have booked a big room as a backup plan!

**Comment by binoculars** [comment]
I'm really happy to see that there is formal discussion around template instantiation and html modules. Thank you all for pushing these forward!

**Comment by rniwa** [comment]
Yeah, I gave my preference to Toronto but Mon-Tue in Tokyo is fine for me.

**Comment by TakayoshiKochi** [comment]
Thanks all, let's fix the place and date: Tokyo, Mon-Tue (Mar. 5-6)!

**Comment by chaals** [comment]
[Meeting page](https://github.com/w3c/WebPlatformWG/blob/gh-pages/meetings/18-03-Web-components.md) is now up.

Please add yourself by PR if you are attending / have agenda requests. We will scrape this issue for the agenda too.

**Comment by GarrettS** [comment]
[CMN: The content of this was inappropriate - an unprofessional ad hominem attack on an individual - and has been deleted]

**Comment by rniwa** [comment]
In the future, please be reminded of [Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/).

I understand your sentiment but that's not a productive feedback. If you have specific concerns or complaints about web components, please file a new issue.

**Comment by treshugart** [comment]
Would it be worth discussing declarative shadow DOM on one of the days?

**Comment by tomalec** [comment]
As Anne [asked for a proposal](https://github.com/w3c/webcomponents/issues/713#issuecomment-347780925). I'll publish my strawman proposal today, maybe someone would be interested in reviewing it.

**Comment by tomalec** [comment]
My slightly more structured proposal for Declarative Shadow DOM is at 
https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Declarative-Shadow-DOM.md (https://github.com/w3c/webcomponents/pull/731)

**Comment by treshugart** [comment]
Nice one, @tomalec. I raised an issue with some of my ideas awhile back in https://github.com/whatwg/dom/issues/531. I've also published some similar musings in https://github.com/treshugart/react-shade which uses React portals to implement a declarative API. I'd be interested in using your proposal as a base to discuss related ideas.

**Comment by treshugart** [comment]
I'll be attending on behalf of SkateJS / Atlassian.

**Comment by travisleithead** [comment]
Per my [previous note](https://github.com/w3c/webcomponents/issues/713#issuecomment-351255513):
> We'll try to have someone from Microsoft be there regardless of the dates--either myself or (hopefully) one or two of my colleagues; haven't figured that out just yet...

Looks like @patrickkettner will be able to be there in person. Thanks Patrick!

**Comment by annevk** [comment]
I think we should do some in-person triaging for these issues:

* Needs implementer interest: https://github.com/w3c/webcomponents/issues?q=is%3Aissue+is%3Aopen+label%3A%22needs+implementer+interest%22
* Needs consensus: https://github.com/w3c/webcomponents/issues?q=is%3Aopen+is%3Aissue+label%3A%22needs+consensus%22

Reducing the number of these issues by stating implementer interest or agreement early is welcome. Then we can use the face-to-face time for other things. (I've mostly removed the v2 label except for things we have clear agreement on. That makes it clearer where the various issues are at in the process.)

Another thing we should discuss is how to address the remaining v1 and upstream issues. In particular there's quite a lot of UI/focus/editing stuff that hasn't really been getting attention while being somewhat essential for the success of shadow trees.

**Comment by rniwa** [comment]
I just reviewed & commented on most issues.

**Comment by TakayoshiKochi** [comment]
The meeting is only 1 week away!

If you attend the meeting but have not put your name on w3c's [minutes](https://github.com/w3c/WebPlatformWG/blob/gh-pages/meetings/18-03-Web-components.md) page, please register yourself by the end of Tuesday this week (Feb. 27) (at your timezone).

We (host at Google Tokyo office) are collecting the list for distributing security badges for the building and the office for the guests, and we need your names for it.

If you have any last minute change on your travel after that, don't worry, please let me (kochi@google.com) know.  We have some room for additional people.

**Comment by annevk** [comment]
@TakayoshiKochi it doesn't seem like https://github.com/w3c/WebPlatformWG/blob/gh-pages/meetings/18-03-Web-components.md#agenda aligns with https://github.com/w3c/webcomponents/issues/713#issue-275960655. I sure hope we keep the unconference aspects to some extent. We could potentially replace it by just asking everyone at the start of the meeting what they think requires discussion most urgently and adjust the agenda based on that. In general we want to keep it pretty flexible I think.

**Comment by TakayoshiKochi** [comment]
@annevk yeah, will update it (I am thinking of this thread is the canonical discussion for the agenda).

**Comment by diervo** [comment]
Would it be possible have a slot to talk about the progress made on the AOM proposal?

**Comment by annevk** [comment]
@diervo yeah, we should definitely discuss accessibility. I think the best thing as I said in an earlier comment is that we each state briefly at the beginning what we want to discuss and make up the agenda from that. And also do some (quick) issue triage together to cover the various proposals that have been made over the years.

**Comment by TakayoshiKochi** [comment]
Thanks all for the productive discussions!

The actual agenda used was:
https://public.etherpad-mozilla.org/p/web_components_agenda

@chaals would you post the meeting minutes at https://github.com/w3c/WebPlatformWG/blob/gh-pages/Meetings.md ?

**Comment by TakayoshiKochi** [comment]
The minutes are posted, thanks @chaals and @diervo !

Monday: https://www.w3.org/2018/03/05-webplat-minutes.html
Tuesday: https://www.w3.org/2018/03/06-webplat-minutes.html

(edit: the second link was wrong, fixed now)

**Comment by rniwa** [comment]
See https://github.com/w3c/webcomponents/issues/786 for 2019's spring F2F.

**Comment by dogoku** [comment]
Hi all, is anyone from the public allowed to attend these meetings?
